aws-analytics-pinpoint/src/main/java/com/amplifyframework/analytics/pinpoint/AmazonPinpointAnalyticsPlugin.java

```diff
@@ -79,8 +96,84 @@ public void enable() {
      * {@inheritDoc}
      */
     @Override
-    public void identifyUser(@NonNull String userId, @NonNull AnalyticsProfile profile) {
-        throw new UnsupportedOperationException("This operation has not been implemented yet.");
+    public void identifyUser(@NonNull String userId, @Nullable UserProfile userProfile) {
```

Comment:
Right now, if `userProfile` is `null`, then it will get passed down into `addUserProfileToEndpoint(...)`, where `userProfile.getSomething()` happens, and will generate a`NullPointerException`.

So, either you should guard `if (userProfile != null) { useIt(); }`, or you should dis-allow `null` by `requireNonNull(...)` immediately after receiving the value.

It's possible that `null` for htis param is supposed to mean "clear the identified user."?

Reply:
Added null check for userProfile. userProfile = null means that the endpoint will have userId as the only information about the user and nothing more.